File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/SkippedTurnEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Event occurring when the bot has skipped a turn, meaning that no intent has reached the server
 * for a specific turn.
 */
public final class SkippedTurnEvent extends BotEvent {

    /**
     * Initializes a new instance of the SkippedTurnEvent class.
     *
     * @param turnNumber is the turn number which was skipped.
     */
    public SkippedTurnEvent(int turnNumber) {
        super(turnNumber);
    }

    /**
     * {@inheritDoc}
     *
     * <p>This event is critical.
     *
     * @return true
     */
    @Override
    public boolean isCritical() {
        return true;
    }
}

File: bot_api_py/events/skipped_turn_event.py
from bot_api_py.events.bot_event import BotEvent


class SkippedTurnEvent(BotEvent):
    """Event occurring when the bot has skipped a turn."""

    def is_critical(self) -> bool:
        """
        Returns:
            True

        This event is critical.
        """
        return True